src/ParallelAlgorithms/Events/ObserveNorms.hpp

```diff
@@ -0,0 +1,372 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+#include <pup.h>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/TagName.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "IO/Observer/Helpers.hpp"
+#include "IO/Observer/ObservationId.hpp"
+#include "IO/Observer/ObserverComponent.hpp"
+#include "IO/Observer/ReductionActions.hpp"
+#include "IO/Observer/TypeOfObservation.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/ArrayIndex.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Reduction.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Event.hpp"
+#include "Time/Tags.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/Numeric.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits/IsA.hpp"
+
+namespace Events {
+/*!
+ * \brief Compute norms of tensors in the DataBox and write them to disk.
+ *
+ * The L2 norm is computed as the RMS, so
+ *
+ * \f{align*}{
+ * L_2(u)=\sqrt{\frac{1}{N}\sum_{i=0}^{N} u_i^2}
+ * \f}
+ *
+ * where \f$N\f$ is the number of grid points.
+ *
+ * The norm can be taken for each individual component, or summed over
+ * components. For the max/min it is then the max/min over all components, while
+ * for the L2 norm we have (for a vector)
+ *
+ * \f{align*}{
+ * L_2(v^i)=\sqrt{\frac{1}{N}\sum_{i=0}^{N} \left[(v^x_i)^2 + (v^y_i)^2
+ *          + (v^z_i)^2\right]}
+ * \f}
+ */
```

Comment:
Pointing the observer to the example yaml in the test as a snippet would be helpful to clarify how to use this observer